mount/mount_windows.go

```diff
@@ -9,15 +16,70 @@ var (
 
 // Mount to the provided target
 func (m *Mount) Mount(target string) error {
```

Comment:
This target is completely unused, but that makes using this function kind of awkward. The concern is that code cannot rely on calling this function to get access to the data through the mount. This could complicate builder use cases which are trying to build off the snapshot pattern. Could this code be unified with the code in `windows/hcsshim.go` which does mounting to provide the mount at a destination path? Doing the equivalent of a mount + bind mount to achieve this would be fine if that is even possible.

Reply:
Agreed it is awkward to not mount at target.

It is not possible to specify the mount location currently, as the layer is mounted as a volume on the host, not at a path on the host. I think it might be possible to mount the resulting volume on the host at a given path, but this would only be delaying solving the same problem again later, as LCOW layers cannot be mounted on the host regardless due to limitations in NTFS. See https://github.com/containerd/containerd/issues/1359 for more context on this.

On Windows, it makes much more sense to get the mounted path, rather than provide a path to mount at. I'm not sure if mounting the volume on the host would fix this for WCOW, but I can attempt if we think it is worth fixing it here before we deal with the LCOW situation.

Reply:
Let's discuss further later, possibly start to include @tonistiigi in our sync ups